core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -45,9 +47,12 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import javax.annotation.concurrent.NotThreadSafe;
+
 /**
  * The abstract base class for {@link ClientStream} implementations.
  */
+@NotThreadSafe
```

Comment:
I am more trying to understand why this class needs the annotation, specifically compared to other non-thread-safe clases. It seems we could put `NotThreadSafe` on every class that wasn't thread-safe, but to what end? I don't want to sign up for adding the annotation en masse. If there is a clear reason why this one is special, then I could be okay with adding it.